sovtokenfees/sovtokenfees/test/catchup/revert/test_revert_fees_batch_2.py

```diff
@@ -0,0 +1,57 @@
+import functools
+
+import pytest
+from sovtokenfees.constants import FEES
+from sovtokenfees.test.catchup.revert.test_revert_fees import get_ppseqno_from_all_nodes, check_batch_ordered
+from sovtokenfees.test.helper import ensure_all_nodes_have_same_data, check_state
+
+from indy_common.constants import ATTRIB
+from plenum.common.startable import Mode
+from plenum.test.delayers import cDelay
+from plenum.test.helper import assertExp
+from plenum.test.stasher import delay_rules
+from plenum.test.test_node import ensureElectionsDone
+from plenum.test.view_change.helper import ensure_view_change
+from stp_core.loop.eventually import eventually
+
+
+@pytest.fixture(scope="module")
+def tconf(tconf):
+    old_max_size = tconf.Max3PCBatchSize
+    old_time = tconf.Max3PCBatchWait
+    tconf.Max3PCBatchSize = 2
+    tconf.Max3PCBatchWait = 1000
+    yield tconf
+
+    tconf.Max3PCBatchSize = old_max_size
+    tconf.Max3PCBatchWait = old_time
+
+
+def test_revert_during_view_change_all_nodes_set_fees(tconf, nodeSetWithIntegratedTokenPlugin,
+                                    fees_set_with_batch, helpers, looper):
+    """
+        Check that SET_FEES transaction will be written after view change when PREPARE quorum for it is reached
+    """
+    nodes = nodeSetWithIntegratedTokenPlugin
+    node_set = [n.nodeIbStasher for n in nodeSetWithIntegratedTokenPlugin]
+
+    _old_pp_seq_no = get_ppseqno_from_all_nodes(nodeSetWithIntegratedTokenPlugin)
+    helpers.general.set_fees_without_waiting({ATTRIB: 3})
+
+    assert _old_pp_seq_no == get_ppseqno_from_all_nodes(nodeSetWithIntegratedTokenPlugin)
+
+    with delay_rules(node_set, cDelay()):
+        # should be changed for auth rule
+        helpers.general.set_fees_without_waiting({ATTRIB: 4})
+        looper.run(eventually(functools.partial(check_batch_ordered, _old_pp_seq_no, nodeSetWithIntegratedTokenPlugin)))
+        ensure_view_change(looper, nodes)
```

Comment:
For now, we don't revert SET_FEES txn, and setting fees to 4 ({ATTRIB: 4}) should be reverted to {ATTRIB: 3}. Fees attribute in StaticFeeReqHandler class, which used into validation txns with fees. This map will be changed on updateState, but not reverted. I think, that we should change 'fees' attribute to property, which should use state. But for now, maybe we should create a test, which should failed now

Reply:
Added such test